src/Tribe/iCal.php

```diff
@@ -588,125 +589,137 @@ protected function get_body( $posts = [] ) {
-				$item[] = 'DTEND;VALUE=' . $type . ':' . $dtend;
-			} else {
-				// Are we using the sitewide timezone or the local event timezone?
-				$timezone_name = $this->get_timezone( $event_post );
-				$timezone = Tribe__Events__Timezones::build_timezone_object( $timezone_name );
+		$full_format = 'Ymd\THis';
+		$utc_format  = 'Ymd\THis\Z';
+		$all_day     = ( 'yes' === get_post_meta( $event_post->ID, '_EventAllDay', true ) );
+		$time        = (object) [
+			'start'    => tribe_get_start_date( $event_post->ID, false, 'U' ),
+			'end'      => tribe_get_end_date( $event_post->ID, false, 'U' ),
+			'modified' => Tribe__Date_Utils::wp_strtotime( $event_post->post_modified ),
+			'created'  => Tribe__Date_Utils::wp_strtotime( $event_post->post_date ),
+		];
+
+		$type   = 'DATE-TIME';
+		$format = $full_format;
+
+		if ( $all_day ) {
+			$type   = 'DATE';
+			$format = 'Ymd';
+		}
+
+		$tzoned = (object) [
+			'start'    => date( $format, $time->start ),
+			'end'      => date( $format, $time->end ),
+			'modified' => date( $utc_format, $time->modified ),
+			'created'  => date( $utc_format, $time->created ),
+		];
+
+		$dtstart = $tzoned->start;
+		$dtend   = $tzoned->end;
 
-				$item[] = 'DTSTART;TZID=' . $timezone->getName() . ':' . $dtstart;
-				$item[] = 'DTEND;TZID=' . $timezone->getName() . ':' . $dtend;
+		if ( 'DATE' === $type ) {
+			// For all day events dtend should always be +1 day.
+			if ( $all_day ) {
+				$dtend = date( $format, strtotime( '+1 day', strtotime( $dtend ) ) );
 			}
 
-			$item[] = 'DTSTAMP:' . date( $full_format, time() );
-			$item[] = 'CREATED:' . $tzoned->created;
-			$item[] = 'LAST-MODIFIED:' . $tzoned->modified;
-			$item[] = 'UID:' . $event_post->ID . '-' . $time->start . '-' . $time->end . '@' . parse_url( home_url( '/' ), PHP_URL_HOST );
-			$item[] = 'SUMMARY:' . $this->replace( strip_tags( $event_post->post_title ) );
+			$item['DTSTART'] = 'DTSTART;VALUE=' . $type . ':' . $dtstart;
+			$item['DTEND']   = 'DTEND;VALUE=' . $type . ':' . $dtend;
+		} else {
+			// Are we using the sitewide timezone or the local event timezone?
+			$timezone_name = $this->get_timezone( $event_post );
+			$timezone      = Tribe__Events__Timezones::build_timezone_object( $timezone_name );
 
-			$content = apply_filters( 'the_content', tribe( 'editor.utils' )->exclude_tribe_blocks( $event_post->post_content ) );
+			$item['DTSTART'] = 'DTSTART;TZID=' . $timezone->getName() . ':' . $dtstart;
+			$item['DTEND']   = 'DTEND;TZID=' . $timezone->getName() . ':' . $dtend;
+		}
 
-			$item[] = 'DESCRIPTION:' . $this->replace( strip_tags( str_replace( '</p>', '</p> ', $content ) ) );
+		$item['DTSTAMP']       = 'DTSTAMP:' . date( $full_format, time() );
+		$item['CREATED']       = 'CREATED:' . $tzoned->created;
+		$item['LAST-MODIFIED'] = 'LAST-MODIFIED:' . $tzoned->modified;
+		$item['UID']           = 'UID:' . $event_post->ID . '-' . $time->start . '-' . $time->end . '@' . parse_url( home_url( '/' ), PHP_URL_HOST );
```

Comment:
:warning: parse_url() is discouraged because of inconsistency in the output across PHP versions; use wp_parse_url() instead.

<sub>hash: c20c9233f97237f967758ad93fbb378e</sub>